codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -925,5 +920,306 @@ final void incrementStreamableBytesForTree(int numBytes) {
+                // EOS = true
+                connection.forEachActiveStream(writeAllocatedBytesVisitor);
+
+                // We must continue writing until the channel is no longer writable. Even though we use an estimate of
+                // how many bytes we can write until the channel becomes unwritable there may be other ChannelHandlers
+                // in the pipeline that reduce the amount of bytes actually written to the channel.
+            } while (connectionState.streamableBytesForTree() > 0 &&
+                     connectionWindowSize > 0 &&
+                     ctx.channel().isWritable());
+        }
+
+        protected final boolean initialWindowSize(int newWindowSize, Http2StreamVisitor writeAllocatedBytesVisitor)
+                throws Http2Exception {
+            if (newWindowSize < 0) {
+                throw new IllegalArgumentException("Invalid initial window size: " + newWindowSize);
+            }
+
+            final int delta = newWindowSize - initialWindowSize;
+            initialWindowSize = newWindowSize;
+            connection.forEachActiveStream(new Http2StreamVisitor() {
+                @Override
+                public boolean visit(Http2Stream stream) throws Http2Exception {
+                    // Verify that the maximum value is not exceeded by this change.
+                    state(stream).incrementStreamWindow(delta);
+                    return true;
+                }
+            });
+
+            if (delta > 0) {
+                // The window size increased, send any pending frames for all streams.
+                writePendingBytes(writeAllocatedBytesVisitor);
+                return false;
+            }
+            return true;
+        }
+
+        @Override
+        public void incrementWindowSize(AbstractState state, int delta) throws Http2Exception {
+            state.incrementStreamWindow(delta);
+        }
+
+        @Override
+        public void enqueueFrame(AbstractState state, FlowControlled frame) throws Http2Exception {
+            state.enqueueFrame(frame);
+        }
+    }
+
+    /**
+     * Provides no notification or tracking of writablity changes.
+     */
+    private final class DefaultWritabilityMonitor extends AbstractWritabilityMonitor {
+        protected final Http2StreamVisitor writeAllocatedBytesVisitor = new Http2StreamVisitor() {
+            @Override
+            public boolean visit(Http2Stream stream) {
+                state(stream).writeAllocatedBytes();
+                return true;
+            }
+        };
+
+        @Override
+        public void writePendingBytes() throws Http2Exception {
+            writePendingBytes(writeAllocatedBytesVisitor);
+        }
+
+        @Override
+        public void initialWindowSize(int newWindowSize) throws Http2Exception {
+            initialWindowSize(newWindowSize, writeAllocatedBytesVisitor);
+        }
+
+        @Override
+        public void pendingBytesChanged(int delta) {
```

Comment:
For no-op methods, maybe add them to the abstract base class and don't override?